neuro_san/service/mcp/processors/mcp_tools_processor.py

```diff
@@ -166,11 +195,17 @@ async def _get_tool_description(self, agent_name: str, metadata: Dict[str, Any])
             }
         }
 
-    async def _extract_tool_response_part(self, response_dict: Dict[str, Any]) -> str:
+    async def _extract_tool_response_part(self, response_dict: Dict[str, Any]) -> Tuple[str, Dict[str, Any]]:
+        """
+        Extract tool response part from the given streaming chat response dictionary.
+        :param response_dict: streaming chat response dictionary;
+        :return: tuple of (text part as string or None, structure part as dictionary or None)
+        """
```

Comment:
We extract both "text" and "structure" parts of AGENT_FRAMEWORK message to be able to build correct response dictionary.